crates/flox/src/commands/general.rs

```diff
@@ -16,83 +15,41 @@ use serde::Serialize;
 use tokio::fs;
 use toml_edit::Key;
 
-use crate::config::features::Feature;
+use crate::commands::not_help;
 use crate::config::{Config, ReadWriteError, FLOX_CONFIG_FILE};
+use crate::subcommand_metric;
 use crate::utils::metrics::{
     METRICS_EVENTS_FILE_NAME,
     METRICS_LOCK_FILE_NAME,
     METRICS_UUID_FILE_NAME,
 };
-use crate::{flox_forward, subcommand_metric};
 
+/// reset the metrics queue (if any), reset metrics ID, and re-prompt for consent
 #[derive(Bpaf, Clone)]
-pub struct GeneralArgs {}
-
-impl GeneralCommands {
-    pub async fn handle(&self, mut config: Config, mut flox: Flox) -> Result<()> {
-        match self {
-            GeneralCommands::Gh(_) => subcommand_metric!("gh"),
-            GeneralCommands::Config(_) => subcommand_metric!("config"),
-            GeneralCommands::ResetMetrics => subcommand_metric!("reset-metrics"),
-            GeneralCommands::Nix(_) => subcommand_metric!("nix"),
+pub struct ResetMetrics {}
+impl ResetMetrics {
+    pub async fn handle(self, _config: Config, flox: Flox) -> Result<()> {
+        subcommand_metric!("reset-metrics");
```

Comment:
before, we were handling metrics submissions in a separate match statement.
I moved these into the handler function now (in preparation to eventually replace them by an [`instrument`](https://docs.rs/tracing/latest/tracing/attr.instrument.html) decoration).

Since still do the feature forwarding at a higher level, feature forwarded calls into bash are not tracked now.
We can now either move the feature handling into the handler methods, or submit metrics in the old match statement'y way for all or the forwarded commands.
What's your preference? @zmitchell @mkenigs

Reply:
done in [d6f7167](https://github.com/flox/flox/pull/215/commits/d6f7167891272682b26dab5b315f2fe7babe839b)
and
[ed94d3a](https://github.com/flox/flox/pull/215/commits/ed94d3ae9075a0abb0e6836f0ac115689ae495ef)
for forwarded commands

Reply:
Looks fine to me how you did it